examples/ChatQnA/deploy/xeon.md

```diff
@@ -0,0 +1,839 @@
+ChatQnA, the following docker images will be needed: embedding, retriever,
+rerank, LLM and dataprep. Additionally, you will need to build docker images for
+ChatQnA megaservice, and UI (conversational React UI is optional). In total,
+there are 8 required and an optional docker images.
+
+The docker images needed to setup the example needs to be build local, however
+the images will be pushed to docker hub soon by Intel.
+
+### Build/Pull Microservice images
+
+From within the `GenAIComps` folder
+
+#### Build Dataprep Image
+
+```
+docker build --no-cache -t opea/dataprep-redis:latest --build-arg https_proxy=$https_proxy \
+  --build-arg http_proxy=$http_proxy -f comps/dataprep/redis/langchain/docker/Dockerfile .
+```
+
+#### Build Embedding Image
+
+```
+docker build --no-cache -t opea/embedding-tei:latest --build-arg https_proxy=$https_proxy \
+   --build-arg http_proxy=$http_proxy -f comps/embeddings/langchain/docker/Dockerfile .
+```
+
+#### Build Retriever Image
+
+```
+docker build --no-cache -t opea/retriever-redis:latest --build-arg https_proxy=$https_proxy \
+   --build-arg http_proxy=$http_proxy -f comps/retrievers/langchain/redis/docker/Dockerfile .
+```
+
+#### Build Rerank Image
+
+```
+docker build --no-cache -t opea/reranking-tei:latest --build-arg https_proxy=$https_proxy \
+   --build-arg http_proxy=$http_proxy -f comps/reranks/tei/docker/Dockerfile .
+```
+
+#### Build docker
+
+::::{tab-set}
+
+:::{tab-item} vllm
+:sync: vllm
+
+We build the vllm docker image from source
+```
+git clone https://github.com/vllm-project/vllm.git
+cd ./vllm/
+docker build --no-cache -t opea/vllm:latest --build-arg https_proxy=$https_proxy \
+   --build-arg http_proxy=$http_proxy -f Dockerfile.cpu .
+cd ..
+```
+
+Next, we'll build the vllm microservice docker. This will set the entry point
+needed for the vllm to suit the ChatQnA examples
+```
+docker build --no-cache -t opea/llm-vllm:latest --build-arg https_proxy=$https_proxy \
+   --build-arg http_proxy=$http_proxy \
+   -f comps/llms/text-generation/vllm/docker/Dockerfile.microservice .
+```
+:::
+:::{tab-item} TGI
+:sync: TGI
+
+```
+docker build --no-cache -t opea/llm-tgi:latest --build-arg https_proxy=$https_proxy \
+   --build-arg http_proxy=$http_proxy -f comps/llms/text-generation/tgi/Dockerfile .
+```
+:::
+::::
+
+### Build Mega Service images
+
+The Megaservice is a pipeline that channels data through different
+microservices, each performing varied tasks. We define the different
+microservices and the flow of data between them in the `chatqna.py` file, say in
+this example the output of embedding microservice will be the input of retrieval
+microservice which will in turn passes data to the reranking microservice and so
+on. You can also add newer or remove some microservices and customize the
+megaservice to suit the needs.
+
+Build the megaservice image for this use case
+
+```
+cd ..
+cd GenAIExamples/ChatQnA/docker
+git checkout tags/v0.9

```

Comment:
Same as above: How is the tag maintained/updated as the released version changes?